### PR TITLE
Bye Bye iOS 9

### LINKED ISSUE #6678: Move base SDK target to 10.0+
We're at 90%+ using iOS 10. Let's drop support for < iOS 10.

### PR DESCRIPTION
**Pull request by astralbodies**
Fixes #6678

To test:
1. Build.
2. Run.
3. Dance.

Needs review: @aerych

### CODE REVIEW

**Comment by aerych** [comment]
Tests perfectly. :)
However, there are oodles of new deprecation warnings with this patch.  It would be nice if we could address them before merging but if there's a reason to handle them in another PR that would be fine too.  Thoughts?

**Comment by astralbodies** [comment]
OMGosh I didn't see any of those warnings.

munch-scream-animation1

Definitely will resolve those in this PR.

**Comment by astralbodies** [comment]
@jleandroperez or @koke Can you check out this branch and see all the notifications changes I had to do? There are two warnings I am not sure the right way to approach.

**Comment by astralbodies** [comment]
@frosty Can you check out the change I made for the split view controller?

**Comment by astralbodies** [comment]
@frosty Oh also maybe you want to rip out `WPNotificationFeedbackGenerator` instead of patching it like I did. 😄

**Comment by astralbodies** [comment]
@SergioEstevao Can you take a look at the changes I made to the API classes? I think we might be able to remove more code.

**Comment by astralbodies** [comment]
Coincidentally I am disappointed that Helpshift's SDK has so many warnings still with iOS 10...

**Comment by koke** [comment]
Pushed bddadae to fix one of the warnings.
The `notificationsEnabledInDeviceSettings` warning is trickier, since the new API is async so that'll take a bit of work.
I also found out we still have some `UIDevice.isOS8` calls, not sure if we want to clean those up here.
Helpshift 😞➕➕

File: Podfile [hunk 1]
```diff
@@ -21,16 +21,16 @@ abstract_target 'WordPress_Base' do
     pod 'AFNetworking',	'3.1.0'
     pod 'CocoaLumberjack', '~> 2.2.0'
     pod 'FormatterKit', '~> 1.8.1'
-    pod 'Helpshift', '~> 5.7.1'
-    pod 'HockeySDK', '~> 3.8.0', :configurations => ['Release-Internal', 'Release-Alpha']
+    pod 'Helpshift', '~> 5.9.3'
```
Comment (koke) [inline Podfile]: + `, :inhibit_warnings => true`? `¯\_(ツ)_/¯`

Comment (koke) [inline Podfile]: Doesn't seem to work 😞

Comment (astralbodies) [inline Podfile]: Yeah it seems to be all or nothing with warnings in pods 😢

**Comment by aerych** [comment]
One remaining warning in `PushNotificationManager.swift`, `'UIUserNotificationType' was deprecated in iOS 10.0: Use UserNotifications Framework's UNAuthorizationOptions` 

```
func notificationsEnabledInDeviceSettings() -> Bool {
        return (sharedApplication.currentUserNotificationSettings?.types ?? UIUserNotificationType()) != UIUserNotificationType()
    }
```
Looks like the tests were updated. Maybe this was also changed but just not added/pushed?

**Comment by koke** [comment]
> One remaining warning in PushNotificationManager.swift

@aerych: @jleandroperez is taking care of that one in #6695

**Comment by aerych** [comment]
Ahh.. Thanks @koke :)

**Comment by astralbodies** [comment]
Thanks for all the eyes on this. 🐻 🤗

**Pull request merged**